test/common/testlib.py

```diff
@@ -654,6 +662,94 @@ def snapshot(self, title, label=None):
             attach(filename)
             print("Wrote HTML dump to " + filename)
 
+    def assert_pixels(self, selector, key, ignore=[]):
+        """Compare the given element with its reference"""
+
+        if Image and self.pixels_label and self.cdp and self.cdp.valid:
+            rect = self.call_js_func('ph_element_clip', selector)
+
+            @dataclass
+            class Rect:
+                x0: int
+                y0: int
+                x1: int
+                y1: int
+
+            def relative_clip(sel):
+                try:
+                    r = self.call_js_func('ph_element_clip', sel)
+                    return Rect(r['x'] - rect['x'],
+                                r['y'] - rect['y'],
+                                r['x'] - rect['x'] + r['width'],
+                                r['y'] - rect['y'] + r['height'])
+                except Error:
+                    return None
+
+            ignore_rects = list(map(relative_clip, ignore))
```

Comment:
Doesn't need an explicit list() conversion, keeping this as iterator is fine.

Reply:
I had trouble without it... the `for` loop below would be empty, for some reason... I'll debug this.

Reply:
Yes, it's confirmed.  Apply this debugging instrumentation:

```
def ignorable_coord(x, y):
    print("IGN", x, y)
    for r in ignore_rects:
        print("on", r)
        if r and x >= r.x0 and x < r.x1 and y >= r.y0 and y < r.y1:
            return True
    return False
```

make a change to a reference image in a ingored part so that ignorable_coord is actually called, and you get this:

```
IGN 230 465
on Browser.assert_pixels.<locals>.Rect(x0=190, y0=459, x1=816, y1=480)
IGN 231 465
IGN 232 465
IGN 233 465
IGN 234 465
IGN 235 465
IGN 236 465
...
Differences in pixel test TestMachinesCreate-testCreateBasicValidation-auto
```

Thus, the first call to ignorable_coords works on the expected rectangle, and then the next calls don't do anything. It is as if the result of `map` can only be consumed once.

Reply:
> It is as if the result of map can only be consumed once.

And that's exactly what's going on, and it is by design. Weird. *shrug*

Reply:
Right, it's an iterator object. Makes sense then, thanks!